gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/CallGraphBugChecker.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.graph.MutableValueGraph;
+import com.google.common.graph.Traverser;
+import com.google.common.graph.ValueGraphBuilder;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.CompilationUnitTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class CallGraphBugChecker extends GradleGuideBugChecker {
+    /**
+     * A directed graph of "who calls who, and where" within this compilation unit.
+     * A compilation unit is just fancy speak for a java file.
+     */
+    public class MethodCallGraph {
+        /**
+         * Nodes are method declarations.
+         * The edge between two methods f and g are all the instances where f calls g directly.
```

Comment:
```suggestion
         * The edge between two methods {@code f} and {@code g} are all the instances where {@code f} calls {@code g} directly.
```

similarly for other identifiers in the javadoc pls

Reply:
yep!